### DISCUSSION TITLE
Docker container with updates using github actions

### DISCUSSION THREAD

**Discussion by danstis** [Kilvoctu/aiyabot#118] [Ideas]
Hi @Kilvoctu are you interested in having the bot packaged into a docker container and published to the docker hub?
I'm happy to create a PR that does all the automatic building if you are interested.

**Comment by Kilvoctu** [comment]
Heya, yes I think I'd be interested in that if it's not too much trouble. I'm not at all familiar with docker, so I had to read up on it and ask someone knowledgeable on the matter 😅, though I'll need to learn more about it still.

**Reply by danstis** [reply]
Ok, I created a PR with this: https://github.com/Kilvoctu/aiyabot/pull/120

Let me know if you want help understanding what I have done. 

Happy to raise another PR if you are interested in SonarCloud scanning of the code base etc. as well.

**Reply by danstis** [reply]
Oh, I forgot to add some documentation to that PR, maybe I should get that added before you merge it?

**Reply by Kilvoctu** [reply]
Yeah, some documentation would be useful, too. I'm looking at the PR 👍👍

Also, I'll look into what SonarCloud is, lol. Thanks for being so helpful; I've been working on this project mostly solo and fairly heavily in my free time for quite a few months, so I'm quite taken aback 😲

**Reply by danstis** [reply]
Yeah cool, I'll update the PR with some documentation, want me to squash the commits into one, or not concerned?

SonarCloud gives you automated code reviews, looks like this: https://sonarcloud.io/project/overview?id=danstis_rmstale
It's free for open-source repos.

**Reply by Kilvoctu** [reply]
It's all right, I'm not concerned about the number of commits.

I did look at SonarCloud a bit, but didn't know it was free for open-source. I'm using PyCharm which has been useful, but if SonarCloud is additionally beneficial I could use all the help I can get haha.

**Reply by danstis** [reply]
Ok, I have updated the PR with some documentation and examples. Let me know if you want something more in there.

**Reply by Kilvoctu** [reply]
Heya, I think it looks all good to me 👍 Is the PR ready to merge?  

Thanks so much again for your help. I can't stress enough my appreciation 👍👍

**Reply by danstis** [reply]
Yep, all good to merge now. 

No problems, happy to give back, as I'm using your bot... Awesome work by the way!

**Comment by danstis** [comment]
Hi @Kilvoctu - Now we have a docker container in the repo, could you enable the display of packages on the main code screen?

This will give you a section that would look like this: 
![image](https://user-images.githubusercontent.com/2137928/221317568-4a4d9156-9707-452d-8d04-924776a321fa.png)

You do this by going to the main code tab on GitHub, then beside the `About` heading there is a little settings cog:
![image](https://user-images.githubusercontent.com/2137928/221317655-b2314a16-6ad3-42fa-9d81-97a498d1cdc4.png)

If you click on that, then check that packages are ticked:
![image](https://user-images.githubusercontent.com/2137928/221317720-1d80bda5-dcdc-44e6-8e84-22d60817fa22.png)

This features gives people another way to check the docker image tags and versions. :)
You can see this in action on my fork: https://github.com/danstis/aiyabot

**Reply by Kilvoctu** [reply]
I was trying out the Docker implementation earlier, using Docker Desktop on Windows. Can't seem to get it to work for me, however. Using `docker run` command provided, I get error:
![image](https://user-images.githubusercontent.com/2993060/221317623-c2da3511-eba2-4aa9-9b8e-c27a797edb52.png)

I've checked online and tried different ways to use the `-v` argument to resolve it, but haven't been successful. What's the correct process?

Also, added the packages now, thanks!

**Reply by danstis** [reply]
Just doing some testing on docker desktop for windows...

**Reply by danstis** [reply]
For the windows docker for desktop, use the relative path mounts as `-v $PWD/aiyabot/outputs:/app/outputs -v $PWD/aiyabot/resources:/app/resources`.

You could also just use full paths to the location as well, like this `-v c:/temp/aiyabot/outputs:/app/outputs -v c:/temp/aiyabot/resources:/app/resources`

Just looking at another issue with permissions, seems when I run the container for the main repo feed it does not start for me, yet the one from my fork does.

**Reply by danstis** [reply]
I'll update the readme content with the $PWD, as it works in Linux too I believe (will just test it there first).

**Reply by Kilvoctu** [reply]
Yeah, I think I did try the relative path mounts before and ran into the permission issue; getting it again now
![image](https://user-images.githubusercontent.com/2993060/221325690-caaa61ff-205d-44a8-8891-08c474bf914a.png)
but wasnt sure where to go from there

Sorry for the trouble 😅

**Reply by danstis** [reply]
Yep, you hit the same thing I just did. I pushed a new PR to fix this one. 
Not sure that is different from my fork to your original repo, but the PR is the correct way of doing things anyhow, so it's hopefully all it needs.

**Reply by danstis** [reply]
@Kilvoctu - If you want to, you can DM me on discord, my username is QuarterEvil#8189

**Reply by Kilvoctu** [reply]
Nice, it seems like all is good now. Thanks for working with me on resolving that. 👍👍 I wanted to get it running on my machine to understand the Docker experience better 

edit: I'll DM you if I have any questions, thanks again
